book_scraping/spiders/book.py

```diff
@@ -0,0 +1,43 @@
+import scrapy
+from scrapy.http import Response
+
+RATINGS = {"One": 1, "Two": 2, "Three": 3, "Four": 4, "Five": 5}
+
+
+class BookSpider(scrapy.Spider):
+    name = "book"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    @staticmethod
+    def _parse_book_info(response: Response) -> dict:
+        book = response.css(".product_page")
+
+        return {
+            "title": book.css(".product_main > h1::text").get(),
+            "price": float(
+                book.css(".price_color::text").get().replace("£", "")
+            ),
+            "amount_in_stock": int(
+                book.css("p.instock::text").re(r"\d+").pop()
+            ),
+            "rating": RATINGS[
+                book.css(".star-rating::attr(class)").get().split().pop()
+            ],
+            "category": response.css(".breadcrumb > li > a[href]::text")
+            .getall()
+            .pop(),
+            "description": response.css(".product_page > p::text").get(),
+            "upc": book.css('th:contains("UPC") + td::text').get(),
+        }
+
+    def parse(self, response: Response, **kwargs) -> dict:
```

Comment:
Are you sure about dict?
https://docs.python.org/3/library/typing.html#typing.Generator